src/rgw/rgw_sal_config.h

```diff
@@ -189,7 +189,7 @@ class ConfigStore {
   /// Create a zone
   virtual int create_zone(const DoutPrefixProvider* dpp,
                           optional_yield y, bool exclusive,
-                          const RGWZoneParams& info,
+                          RGWZoneParams& info,
```

Comment:
i'd like to avoid making this reference mutable so the `ConfigStore` interface stays as dumb as possible. it should just read/write zone/zonegroup/realm objects as they are

if we really do need different `ConfigStore` types to choose different default `sal_config`, we could do that 'above the line' with string comparisons like `if (conf->rgw_config_store == "dbstore")` or whatever. but i'm starting to think we should leave "rados" as the default for everything

the only thing that needs to be smart or opinionated is the deployment logic, not rgw itself. the deployment tooling can pass in its own sal config, whether that's with `radosgw-admin zone create` or `zone modify`. if you're concerned about the developer experience, vstart.sh would be a good place to set up special configurations for `--dbstore` or `--posix`